follower/consensus_follower.go

```diff
@@ -22,11 +22,12 @@ type ConsensusFollower interface {
 
 // Config contains the configurable fields for a `ConsensusFollower`.
 type Config struct {
-	nodeID               flow.Identifier // the node ID of this node
-	upstreamAccessNodeID flow.Identifier // the node ID of the upstream access node
-	bindAddr             string          // address to bind on
-	dataDir              string          // directory to store the protocol state
-	bootstrapDir         string          // path to the bootstrap directory
+	nodeID                  flow.Identifier // the node ID of this node
+	bootstrapNodeAddresses  []string        // the addresses of the boostrap peers
```

Comment:
@smnzhu changed this to accept the addresses and keys

Reply:
I think it would make for a cleaner external API if instead of two separate arrays, we pass in a single array `bootstrapNodes` of struct `BootstrapNodeInfo`s which contains the address and key. I also think the `BootstrapNodeInfo` should look like this:

```go
type BootstrapNodeInfo struct {
        Host string // ip or hostname
        Port uint 
        NetworkPublicKey crypto.PublicKey
}
```

This makes it very explicit what format the information should be provided in by the consumer of the API, whereas strings are very ambiguous and confusing for the user.

Then below, we would not call `access.BootstrapIdentities` but instead some other function to directly create the identity list, and hence avoiding the need to return any error (`access.BootstrapIdentities` potentially returns an error due to unmarshalling the string of the network key, but this is no longer necessary).